lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1865,16 +1869,21 @@ def _resolve_depenency_map(
             max_rounds=2000000,  # NOTE: same constant pip uses
         ).mapping
     except CollectionDependencyResolutionImpossible as dep_exc:
-        conflict_causes = (
-            '* {req.fqcn!s}:{req.ver!s} ({dep_origin!s})'.format(
-                req=req_inf.requirement,
-                dep_origin='direct request'
-                if req_inf.parent is None
-                else 'dependency of {parent!s}'.
-                format(parent=req_inf.parent),
-            )
-            for req_inf in dep_exc.causes
-        )
+        conflict_causes = []
+        for req_inf in dep_exc.causes:
+            if req_inf.requirement.type == "requires_ansible":
+                collection = f"{req_inf.parent.fqcn!s}:{req_inf.parent.ver!s}"
+                dep_origin = 'direct request' if req_inf.parent._parent is None else f'dependency of {req_inf.parent._parent!s}'
```

Comment:
Can we just check if `parent` is of type `requires_ansible`? I'd expect that to be enough. Does anything prevent this from working? If yes, maybe there's something to be done about this in the resolver hooks.

Reply:
I think I was trying to build a mental model of the changes you're proposing here and didn't understand the context fully. I probably confused the parent-child relationships originally and felt like there's something unnecessarily complicated happening here. But I've lost that context since. The more I'm reading this, the more I realize that just staring at the diff is not enough for me to give useful feedback, and I really do need to inspect this locally.

Reply:
@webknjaz Have you been able to take a closer look? I just rebased this branch since there was a merge conflict. Only the last couple commits are new since you last reviewed.